### PR TITLE
Some block updates

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because there is no stable version.

## Changelog

```markdown
### Added
- Added support for new `EditableBlockService`
- Add missing translation for breadcrumbs

### Changed
- Add `internal` as default context for technical blocks
```

### CODE REVIEW

File: src/Resources/views/BlockAdmin/compose_preview.html.twig [hunk 1]
```diff
@@ -8,13 +8,18 @@
        href="{{ admin_pool.getAdminByAdminCode('sonata.dashboard.admin.block').generateUrl('edit', { 'id': child.id, 'composer': true }) }}"
     >
         {% set service = attribute(blockServices, child.type) %}
+        {% if service.metadata is defined %}
+            {% set metadata = service.metadata %}
+        {% else %}
+            {% set metadata = service.blockMetadata %}
+        {% endif %}
```
Comment (greg0ire) [inline src/Resources/views/BlockAdmin/compose_preview.html.twig]: `{% set metadata = service.metadata|default(service.blockMetadata) %}`?

File: src/Resources/views/DashboardAdmin/compose_container_show.html.twig [hunk 2]
```diff
@@ -18,7 +18,10 @@
     <div class="dashboard-composer__block-type-selector">
         <label>{{ 'composer.block.add.type'|trans({}, 'SonataDashboardBundle') }}</label>
         <select class="dashboard-composer__block-type-selector__select" style="width: auto">
-            {% for blockServiceId, blockService in blockServices %}
+            {% for blockServiceId, blockService in blockServices|filter(service => service.metadata is defined) %}
```
Comment (phansys) [inline src/Resources/views/DashboardAdmin/compose_container_show.html.twig]: The `filter` filter requires `twig/twig:^1.41|^2.10`: https://twig.symfony.com/doc/2.x/filters/filter.html#filter
https://github.com/sonata-project/SonataDashboardBundle/blob/72fd3a1d1825676a1bf8d4c079aabf9b04129690/composer.json#L38

File: src/Admin/BlockAdmin.php [hunk 3]
```diff
@@ -378,4 +432,42 @@ protected function configureFormFields(FormMapper $formMapper): void
...
         }
     }
+
+    private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
+    {
+        $service = $this->blockManager->get($block);
+        $blockType = $block->getType();
+
+        if (!$service instanceof BlockServiceInterface) {
+            throw new \RuntimeException(sprintf(
```
Comment (phansys) [inline src/Admin/BlockAdmin.php]: IMO, `UnexpectedValueException` or `DomainException` seem more accurate for this case.

File: src/Resources/views/BlockAdmin/compose_preview.html.twig (hunk 1 above)
Comment (phansys) [reply src/Resources/views/BlockAdmin/compose_preview.html.twig]: I did the same suggestion on previous PRs, but I don't remember which was the problem preventing this expression.

File: src/Admin/BlockAdmin.php (hunk 3 above)
Comment (core23) [reply src/Admin/BlockAdmin.php]: This is just a copy & paste of https://github.com/sonata-project/SonataPageBundle/pull/1064/files?file-filters%5B%5D=.php

File: src/Resources/views/BlockAdmin/compose_preview.html.twig (hunk 1 above)
Comment (core23) [reply src/Resources/views/BlockAdmin/compose_preview.html.twig]: This is not possible :(
https://github.com/sonata-project/SonataPageBundle/pull/1064

File: src/Resources/views/DashboardAdmin/compose_container_show.html.twig (hunk 2 above)
Comment (core23) [reply src/Resources/views/DashboardAdmin/compose_container_show.html.twig]: Fixed

File: src/Resources/translations/SonataDashboardBundle.en.xliff [hunk 4]
```diff
@@ -290,6 +290,30 @@
...
+                <source>breadcrumb.link_dashboard_block_delete</source>
+                <target>Delete</target>
+            </trans-unit>
+            <trans-unit id="76" resname="breadcrumb.link_dashboard_block_edit">
+                <source>breadcrumb.link_dashboard_block_edit</source>
+                <target>Edit</target>
+            </trans-unit>
+            <trans-unit id="77" resname="breadcrumb.link_dashboard_block_list">
+                <source>breadcrumb.link_dashboard_block_list</source>
+                <target>Blocks list</target>
```
Comment (phansys) [inline src/Resources/translations/SonataDashboardBundle.en.xliff]: Shouldn't this be a verb like the other actions? Mean, "List blocks" or just "List".

Comment (core23) [reply src/Resources/translations/SonataDashboardBundle.en.xliff]: No, it's just the list of all blocks :trollface: 

It's the same like this: https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/Resources/translations/SonataPageBundle.en.xliff#L25-L28

Comment (phansys) [reply src/Resources/translations/SonataDashboardBundle.en.xliff]: I guess I understand, maybe this is a discussion for a more specific PR. But just to share my point of view, let me explain my reasoning.
There are other translations previous this one, expressed in the form of a verb which trigger an action.
IE:
* **Edit** (verb) link will redirect the user to the **edit form** (subject)
* **Delete** (verb) link will redirect the user to the **delete confirmation form** (subject)

So, to my understanding, the "list" action should be expressed in the same way:
* **List** (verb) link will redirect the user to the **blocks list** (subject)

Comment (core23) [reply src/Resources/translations/SonataDashboardBundle.en.xliff]: Changed this to `Blocks`

**Comment by greg0ire** [comment]
Thanks @core23 !

**Pull request merged**